Mathlib/LinearAlgebra/Dual/Lemmas.lean

```diff
@@ -141,106 +146,102 @@ instance dual_finite [Projective R M] : Module.Finite R (Dual R M) :=
+variable [CommSemiring K] [AddCommMonoid V] [Module K V] [Projective K V]
 
 open Module Module.Dual Submodule LinearMap Cardinal Basis Module
 
 section
 
-variable (K) (V)
+variable (K)
 
-theorem eval_ker : LinearMap.ker (eval K V) = ⊥ :=
+theorem eval_apply_injective : Function.Injective (eval K V) :=
   have ⟨s, hs⟩ := Module.projective_def'.mp ‹Projective K V›
-  ker_eq_bot.mpr <| .of_comp (f := s.dualMap.dualMap) <| (ker_eq_bot.mp <|
-    Finsupp.basisSingleOne (R := K).eval_ker).comp (injective_of_comp_eq_id s _ hs)
+  .of_comp (f := s.dualMap.dualMap)
+    (Finsupp.basisSingleOne.eval_injective.comp <| injective_of_comp_eq_id s _ hs)
+
+variable (V)
 
-theorem map_eval_injective : (Submodule.map (eval K V)).Injective := by
-  apply Submodule.map_injective_of_injective
-  rw [← LinearMap.ker_eq_bot]
-  exact eval_ker K V
+theorem eval_ker : LinearMap.ker (eval K V) = ⊥ := ker_eq_bot_of_injective (eval_apply_injective K)
 
-theorem comap_eval_surjective : (Submodule.comap (eval K V)).Surjective := by
-  apply Submodule.comap_surjective_of_injective
-  rw [← LinearMap.ker_eq_bot]
-  exact eval_ker K V
+theorem map_eval_injective : (Submodule.map (eval K V)).Injective :=
+  Submodule.map_injective_of_injective (eval_apply_injective K)
+
+theorem comap_eval_surjective : (Submodule.comap (eval K V)).Surjective :=
+  Submodule.comap_surjective_of_injective (eval_apply_injective K)
 
 end
 
 section
 
 variable (K)
 
-theorem eval_apply_eq_zero_iff (v : V) : (eval K V) v = 0 ↔ v = 0 := by
-  simpa only using SetLike.ext_iff.mp (eval_ker K V) v
-
-theorem eval_apply_injective : Function.Injective (eval K V) :=
-  (injective_iff_map_eq_zero' (eval K V)).mpr (eval_apply_eq_zero_iff K)
+theorem eval_apply_eq_zero_iff (v : V) : (eval K V) v = 0 ↔ v = 0 :=
+  SetLike.ext_iff.mp (eval_ker K V) v
 
 theorem forall_dual_apply_eq_zero_iff (v : V) : (∀ φ : Module.Dual K V, φ v = 0) ↔ v = 0 := by
   rw [← eval_apply_eq_zero_iff K v, LinearMap.ext_iff]
   rfl
 
 @[simp]
-theorem subsingleton_dual_iff :
-    Subsingleton (Dual K V) ↔ Subsingleton V := by
-  refine ⟨fun h ↦ ⟨fun v w ↦ ?_⟩, fun _ ↦ inferInstance⟩
-  rw [← sub_eq_zero, ← forall_dual_apply_eq_zero_iff K (v - w)]
-  intros f
-  simp [Subsingleton.elim f 0]
+theorem subsingleton_dual_iff : Subsingleton (Dual K V) ↔ Subsingleton V :=
+  ⟨fun _ ↦ ⟨fun _ _ ↦ eval_apply_injective K (Subsingleton.elim ..)⟩, fun _ ↦ inferInstance⟩
 
 @[simp]
-theorem nontrivial_dual_iff :
-    Nontrivial (Dual K V) ↔ Nontrivial V := by
+theorem nontrivial_dual_iff : Nontrivial (Dual K V) ↔ Nontrivial V := by
   rw [← not_iff_not, not_nontrivial_iff_subsingleton, not_nontrivial_iff_subsingleton,
     subsingleton_dual_iff]
 
 instance instNontrivialDual [Nontrivial V] : Nontrivial (Dual K V) :=
   (nontrivial_dual_iff K).mpr inferInstance
 
 omit [Projective K V] in
+/- May fail if `V` is only a projective `R`-module;
+see https://stacks.math.columbia.edu/tag/05WG#comment-9913. -/
```

Comment:
By "may fail if" I assume you mean "May not generalize to the case when"?

Reply:
I would say "does not generalize to the case when".